book/src/engine.md

```diff
@@ -1,3 +1,23 @@
-# How does the engine work
+# Chalk engine
 
-*TBD:* Explain how chalk-engine 
+The `chalk-engine` crate is the core PROLOG-like solver for logical
```

Comment:
BTW, have you seen the [docs on the SLG solver in the rustc-guide](https://rust-lang.github.io/rustc-guide/traits/slg.html)? I've not reviewed them recently :) but we may want to incorporate some ideas from them, not sure.

Reply:
I have seen them, but haven't looked at them in a bit. I think it's a good idea to incorporate them, if there are things not covered once we've written more of the book? I could go through and see if there's anything worth adding to *this* chapter, but stretched for time at the moment, and would rather see what *is* here landed, tbh.